lib/archethic/contracts/interpreter/library.ex

```diff
@@ -56,6 +116,65 @@ defmodule Archethic.Contracts.Interpreter.Library do
     end
   end
 
+  @doc """
+  Extract data from string using capture groups
+  (multiline flag is activated)
+
+  ps: the number of antislash is doubled because this is a doctest
+
+  ## Examples
+
+      iex> Library.regex_scan("foo", "bar")
+      []
+
+      iex> Library.regex_scan("toto,123\\ntutu,456\\n", "toto,([0-9]+)")
+      ["123"]
+
+      iex> Library.regex_scan("toto,123\\ntutu,456\\n", "t.t.,([0-9]+)")
+      ["123", "456"]
+
+      iex> Library.regex_scan("A0B1C2,123\\nD3E4F5,456\\n", "^(\\\\w+),(\\\\d+)$")
+      [["A0B1C2", "123"], ["D3E4F5", "456"]]
+
+  """
+  @spec regex_scan(binary(), binary()) :: list(binary())
+  def regex_scan(text, pattern) when is_binary(text) and is_binary(pattern) do
+    case Regex.compile(pattern, "m") do
+      {:ok, pattern} ->
+        Regex.scan(pattern, text, capture: :all_but_first)
+        |> Enum.map(fn
```

Comment:
You can use `Enum.flat_map(& &1)`

Reply:
I don't want to flatten everything, I only want to flat if there's one capture group, similar behaviour as `regex_extract`.